### PR TITLE
fix: outbound tests with exclude

### PR DESCRIPTION
**Pull request by slonka**
Signed-off-by: slonka <slonka@users.noreply.github.com>

The exclude tests never worked. This PR fixes them by exposing the server on another namespace.

Did you sign your commit? [Instructions](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md#sign-your-commits) yes

Have you read [Contributing guidelines](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md)? yes

### CODE REVIEW

**Comment by slonka** [comment]
This shows we have an error in exclude ports tests or the functionality itself

File: test/blackbox_tests/outbound_redirect_test.go [hunk 1]
```diff
@@ -175,17 +174,28 @@ var _ = Describe("Outbound IPv6 TCP traffic to any address:port", func() {
...
 
-var _ = Describe("Outbound IPv4 TCP traffic to any address:port except excluded ones", func() {
+var _ = FDescribe("Outbound IPv4 TCP traffic to any address:port except excluded ones", func() {
 	var err error
 	var ns *netns.NetNS
+	var ns2 *netns.NetNS
 
 	BeforeEach(func() {
-		ns, err = netns.NewNetNSBuilder().Build()
+		veth := netns.NewVeth("shared-", 1, 2)
```
Comment (slonka) [inline test/blackbox_tests/outbound_redirect_test.go:183]: TODO: use `genSuffixes`

File: test/blackbox_tests/outbound_redirect_test.go [hunk 2]
```diff
@@ -175,17 +174,28 @@ var _ = Describe("Outbound IPv6 TCP traffic to any address:port", func() {
...
+		Expect(netlink.LinkAdd(veth)).To(Succeed())
+		mainLink, err2 := netlink.LinkByName(veth.Name)
+		Expect(err2).To(BeNil())
+		peerLink, err2 := netlink.LinkByName(veth.PeerName)
+		Expect(err2).To(BeNil())
+
+		ns, err = netns.NewNetNSBuilder().WithSharedLink(mainLink).WithLinkAddress("10.255.0.1/24").Build()
```
Comment (slonka) [inline test/blackbox_tests/outbound_redirect_test.go:190]: how do we pick the IP here?

File: test/framework/netns/builder.go [hunk 3]
```diff
@@ -319,6 +333,10 @@ func (b *Builder) Build() (*NetNS, error) {
 			done <- fmt.Errorf("cannot set the default route: %s", err)
 		}
 
+		if err := netlink.RouteAdd(&netlink.Route{LinkIndex: b.sharedLink.Attrs().Index, Dst: &net.IPNet{IP: net.ParseIP("10.254.0.0"), Mask: net.CIDRMask(24, 32)}}); err != nil {
```
Comment (slonka) [inline test/framework/netns/builder.go:336]: make `10.254.0.0` configurable, right?

**Review by lukidzi — APPROVED** [review]

**Comment by slonka** [comment]
related: https://github.com/kumahq/kuma-net/issues/84

**Pull request merged**